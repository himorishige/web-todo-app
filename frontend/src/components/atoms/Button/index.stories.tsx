/** @jsxImportSource @emotion/react */
import { ComponentStory, ComponentMeta } from '@storybook/react';
import Button from '.';

export default {
  title: 'WebToDoApp/Atoms/Button',
  component: Button,
} as ComponentMeta<typeof Button>;

const Template: ComponentStory<typeof Button> = (args) => <Button {...args} />;

export const Primary = Template.bind({});
Primary.storyName = 'Primary';
Primary.args = {
  primary: true,
  label: '登録する',
  disabled: false,
};

export const Secondary = Template.bind({});
Secondary.storyName = 'Secondary';
Secondary.args = {
  label: '削除する',
  disabled: false,
};
