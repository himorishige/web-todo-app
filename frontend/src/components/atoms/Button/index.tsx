/** @jsxImportSource @emotion/react */
import { css } from '@emotion/react';

export type Props = {
  primary?: boolean;
  size?: 'small' | 'medium' | 'large';
  label: string;
  disabled?: boolean;
  onClick?: () => void;
};

const Button: React.VFC<Props> = ({
  primary = false,
  size = 'medium',
  label,
  disabled = false,
  ...props
}: Props) => {
  return (
    <button
      css={[buttonStyle, primary ? primaryStyle : secondaryStyle]}
      disabled={disabled}
      {...props}
    >
      {label}
    </button>
  );
};

export default Button;

const buttonStyle = css`
  padding: 0.5rem 1rem;
  border: none;
  border-radius: 0.25rem;
  cursor: pointer;
  background: var(--secondary-color);
  transition: opacity 0.3s ease-in-out;
  &:hover {
    opacity: 0.8;
  }
  &:disabled {
    cursor: not-allowed;
    background-color: #c5c5c5;
    &:hover {
      opacity: 1;
    }
  }
`;

const primaryStyle = css`
  color: white;
  background: var(--primary-color);
`;

const secondaryStyle = css``;
