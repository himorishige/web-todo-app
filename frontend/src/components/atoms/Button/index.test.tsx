import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Button from '.';
import { matchers } from '@emotion/jest';

// Add the custom matchers provided by '@emotion/jest'
expect.extend(matchers);

describe('Atoms/Button', () => {
  test('ラベル名が正しく表示される', () => {
    render(<Button label="button" />);
    const element = screen.getByRole('button');
    expect(element).toHaveTextContent('button');
  });

  test('primaryボタンの場合色が正しく反映される', () => {
    render(<Button primary label="button" />);
    const element = screen.getByRole('button');
    expect(element).toHaveStyle('background: var(--primary-color)');
  });

  test('正しくレンダリングされている', () => {
    const target = renderer.create(<Button label="button" />).toJSON();
    expect(target).toMatchSnapshot();
  });
});
