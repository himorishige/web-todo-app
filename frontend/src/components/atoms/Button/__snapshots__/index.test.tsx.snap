// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/Button 正しくレンダリングされている 1`] = `
<button
  className="css-qckj2w"
  disabled={false}
>
  button
</button>
`;
